modules/collect/src/main/java/com/opengamma/collect/timeseries/LocalDateDoubleTimeSeries.java

```diff
@@ -1,293 +1,106 @@
-  }
-
-  /**
-   * Obtains a time-series from matching arrays of dates and values.
-   * <p>
-   * The two arrays must be the same size and must be sorted from earliest to latest.
-   *
-   * @param dates  the date list
-   * @param values  the value list
-   * @return the time-series
-   */
-  public static LocalDateDoubleTimeSeries of(Collection<LocalDate> dates, Collection<Double> values) {
-    ArgChecker.noNulls(dates, "dates");
-    ArgChecker.noNulls(values, "values");
-    LocalDate[] datesArray = dates.toArray(new LocalDate[dates.size()]);
-    double[] valuesArray = Doubles.toArray(values);
-    validate(datesArray, valuesArray);
-    return createUnsafe(datesArray, valuesArray);
-  }
-
-  /**
-   * Obtains a time-series from a map of dates and values.
-   *
-   * @param map  the map of dates to values
-   * @return the time-series
-   */
-  public static LocalDateDoubleTimeSeries of(Map<LocalDate, Double> map) {
-    ArgChecker.noNulls(map, "map");
-    Set<Entry<LocalDate, Double>> set = map.entrySet();
-    LocalDate[] datesArray = new LocalDate[set.size()];
-    double[] valuesArray = new double[set.size()];
-    int i = 0;
-    for (Entry<LocalDate, Double> entry : set) {
-      datesArray[i] = entry.getKey();
-      valuesArray[i] = entry.getValue();
-      i++;
-    }
-    validate(datesArray, valuesArray);
-    return createUnsafe(datesArray, valuesArray);
-  }
-
-  /**
-   * Obtains a time-series from a collection of points.
-   * <p>
-   * The collection must be sorted from earliest to latest.
-   *
-   * @param points  the list of points
-   * @return the time-series
-   */
-  public static LocalDateDoubleTimeSeries of(Collection<LocalDateDoublePoint> points) {
-    ArgChecker.noNulls(points, "points");
-    LocalDate[] datesArray = points.stream()
-        .map(LocalDateDoublePoint::getDate)
-        .toArray(LocalDate[]::new);
-    double[] valuesArray = points.stream()
-        .mapToDouble(LocalDateDoublePoint::getValue)
-        .toArray();
-    validate(datesArray, valuesArray);
-    return createUnsafe(datesArray, valuesArray);
-  }
-
-  // creates time-series by directly assigning the input arrays
-  // must only be called when safe to do so
-  private static LocalDateDoubleTimeSeries createUnsafe(LocalDate[] dates, double[] values) {
-    return new LocalDateDoubleTimeSeries(dates, values, true);
-  }
-
-  // validates the arrays are same length and in order
-  private static void validate(LocalDate[] dates, double[] values) {
-    ArgChecker.isTrue(dates.length == values.length,
-        "Arrays are of different sizes - dates: {}, values: {}", dates.length, values.length);
-    LocalDate maxDate = LocalDate.MIN;
-    for (LocalDate date : dates) {
-      ArgChecker.isTrue(date.isAfter(maxDate),
-          "Dates must be in ascending order but: {} is not after: {}", date, maxDate);
-      maxDate = date;
-    }
-  }
+public interface LocalDateDoubleTimeSeries extends Bean {
```

Comment:
Other interfaces extend ImmutableBean. They have this at the end of the class Javadoc:

 * <p>
 * Implementations must be immutable and thread-safe beans.